### DISCUSSION TITLE
Auto-reload when markdown files are changed

### DISCUSSION THREAD

**Discussion by jylin** [vercel/next.js#11419] [Help]
For the blog-starter example, in dev mode, is there any way to make the page auto reload when md files are changed like how it auto reloads when any of the javascript/typescript files are changed?

**Comment by urish** [comment]
I'm also looking at this. I have a similar setup for [GoodArduinoCode](https://goodarduinocode.com), where the actual content lives inside markdown files, and I'd like it to be automatically reloaded whenever the markdown files changes. Ideally, this would happen through fast-refresh, but even a standard refresh will do the trick for me...

**Reply by urish** [reply]
Eventually, I implemented this using a new API endpoint that returns the current version of the files + polling with setInterval: https://github.com/wokwi/good-arduino-code/commit/839400af07a67420dcb2f75b8226ef482f4368cb

**Reply by thomvaill** [reply]
Hey! Thank you for your answer!
It inspired me a lot for my [Log4brains project](https://github.com/thomvaill/log4brains/), in which I had to develop a markdown local preview with a "hot reload" system.

I've improved your solution a little to:

- Improve responsiveness
- Benefit from Next's own refresh system, without having to update the app state by hand

Therefore, I've opted for a push-based solution with Socket.io and chokidar for file watching.
Which triggers a simple `await Router.replace(window.location.href);` on the client side, that refreshes the page without a full reload.

However, a small hack is required to tell Next not to use its client-side cache system, coupled with the `revalidate: 1` page option:

```typescript
async function hotReloadCurrentPage(): Promise<void> {
  /**
   * #NEXTJS-HACK
   * We clear Next.JS Router's "static data cache" to make our Hot Reload feature work.
   * So that we force a new request to the server.
   */
  // @ts-ignore
  Router.router.sdc = {};

  await Router.replace(window.location.href);
}
```

Here are the relevant code snippets:

- Server-side (file watcher with chokidar): https://github.com/thomvaill/log4brains/blob/master/packages/core/src/infrastructure/file-watcher/FileWatcher.ts and https://github.com/thomvaill/log4brains/blob/master/packages/web/src/cli/preview.ts#L44
- Client-side: https://github.com/thomvaill/log4brains/blob/master/packages/web/src/layouts/AdrBrowserLayout/ConnectedAdrBrowserLayout.tsx#L128

Thanks to this solution, the refresh is almost instant!

**Comment by ggrossetie** [comment]
I'm also interested in such feature.
Any suggestions on how it should be implemented?

**Reply by urish** [reply]
I ended up implementing it by polling + API endpoint: 

https://github.com/wokwi/good-arduino-code/commit/839400af07a67420dcb2f75b8226ef482f4368cb

**Reply by ggrossetie** [reply]
Thanks for sharing, I will give it a try :+1:

**Comment by bluprince13** [comment]
I used https://github.com/hashicorp/next-remote-watch to solve this for my blog where I wanted hot reloading for my markdown files.

My commit: https://github.com/bluprince13/bluprince13-next/commit/941c06ecdfce85e47b09a728ae89443a7a753619

**Reply by HerbCaudill** [reply]
+1 for next-remote-watch - works great

**Reply by janigowski** [reply]
If you stuck, make sure the correct folder is watched for changes. You can pass dirpath as first argument: 
```
next-remote-watch posts
```
